doc/security/security-overview.rst

```diff
@@ -380,17 +380,18 @@ The quality assurance part encompasses the following criteria:
    independent reviewer other than the author(s) of the code change.
    These reviews shall be performed by the subsystem maintainers and
    developers on a functional level and are to be distinguished from
-   security reviews as laid out in Chapter 4. Please refer to the
-   `development model documentation`_ on the Zephyr project Wiki.
+   security reviews as laid out in the `Secure Design`_ section.
+   Please refer to the `development model documentation`_ on the
```

Comment:
The dev model docs are now part of the technical docs, so let's change this to:
```
Refer to the :ref:`development_model` documentation for more information.
```

Reply:
@dbkinder Pushed an update for this